gui/control_main.py

```diff
@@ -4072,25 +4104,47 @@ def addRequestsToAllSelectedCB(self):
                     samplesConsidered.add(self.selectedSampleID)
         else:  # If queue collect is off does not matter how many requests you select only one will be added to current pin
             self.selectedSampleID = self.mountedPin_pv.get()
-            self.selectedSampleRequest = daq_utils.createDefaultRequest(
-                self.selectedSampleID
-            )
-            self.dataPathGB.setFilePrefix_ledit(
-                str(self.selectedSampleRequest["request_obj"]["file_prefix"])
-            )
-            self.dataPathGB.setDataPath_ledit(
-                str(self.selectedSampleRequest["request_obj"]["directory"])
-            )
-            self.EScanDataPathGB.setFilePrefix_ledit(
-                str(self.selectedSampleRequest["request_obj"]["file_prefix"])
-            )
-            self.EScanDataPathGB.setDataPath_ledit(
-                str(self.selectedSampleRequest["request_obj"]["directory"])
-            )
-            self.addSampleRequestCB(selectedSampleID=self.selectedSampleID)
+            sample_data = db_lib.getSampleByID(self.selectedSampleID)
+            prop_dir = self.get_proposal_directory(sample_data["proposalID"])
+            add_request = True
```

Comment:
could you extract the `add_request` calculation to outside of this code without breaking up the flow when reading the code?

Reply:
Moved to separate method in 330a9f9d014e950048771cf79166f8df996ba26d